src/hooks/useApi.tsx

```diff
@@ -1,50 +1,92 @@
-import { useState, useEffect } from 'react';
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+
+export interface Item {
+  id?: number;
+  name: string;
+  amount: string;
+  url?: string;
+  tags: string;
+  type?: string;
+  status?: string;
+  deadline?: string;
+  description?: string;
+  isRepetable?: boolean;
+}
+
+const fetchData = async (endpoint: string): Promise<Item[]> => {
+  const response = await fetch(endpoint);
+  if (!response.ok) throw new Error(`Error: ${response.status}`);
+  return response.json();
+};
+
+const addItem = async (endpoint: string, item: Item) => {
+  const response = await fetch(endpoint, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(item),
+  });
+  if (!response.ok) throw new Error(`Error: ${response.status}`);
+  return response.json();
+};
+
+const deleteItem = async (endpoint: string, id: number) => {
+  const response = await fetch(`${endpoint}/${id}`, {
+    method: "DELETE",
+  });
+  if (!response.ok) throw new Error(`Error: ${response.status}`);
+  return id;
+};
+
+const updateItem = async (endpoint: string, id: number, updatedItem: Item) => {
+  const response = await fetch(`${endpoint}/${id}`, {
+    method: "PUT",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(updatedItem),
+  });
+  if (!response.ok) throw new Error(`Error: ${response.status}`);
+  return updatedItem;
+};
 
 const useApi = (endpoint: string) => {
-  const [data, setData] = useState<any[]>([]);
-
-  useEffect(() => {
-    fetchData();
-  }, [endpoint]);
-
-  const fetchData = async () => {
-    try {
-      const response = await fetch(endpoint);
-      if (!response.ok) throw new Error(`Error: ${response.status}`);
-      const result = await response.json();
-      setData(result);
-    } catch (err: any) {
-      console.error(err.message || 'Something went wrong when trying to fetch data');
-    }
-  };
+  const queryClient = useQueryClient();
 
-  const addItem = async (item: any) => {
-    try {
-      const response = await fetch(endpoint, {
-        method: 'POST',
-        headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify(item),
-      });
-      if (!response.ok) throw new Error(`Error: ${response.status}`);
-      await fetchData();
-    } catch (err: any) {
-      console.error(err.message || 'Something went wrong when trying to add item');
-    }
-  };
+  const { data, isLoading } = useQuery<Item[], Error>({
```

Comment:
Notice that the fetch is being executed on every page navigation (=everywhere you use this hook). This is fine for pages where we use the items, since we want to fetch the latest. However, the add item page does do anything with existing items, so the api request is unnecessary additional overhead.

To prevent this, either separate the reads into different hooks, or perhaps disable the query conditionally via a prop to this hook.